### PR TITLE
Adding bottom margin on .middle-content container to give space between content and browser edge. And address log-viewer white bg

### PR DESCRIPTION
**Pull request by sg00dwin**
Fixes openshift#994
Fixes openshift#1002
Fixes openshift#993

Added `.gutter-bottom.gutter-bottom-2x()` to `.middle-content` since there was no rule that applied default spacing within the the main content panel. And removed it from annotations that was no longer needed.

screen shot 2016-12-07 at 5 06 19 pm

screen shot 2016-12-07 at 5 22 16 pm

screen shot 2016-12-07 at 5 21 11 pm

screen shot 2016-12-07 at 5 20 57 pm

screen shot 2016-12-07 at 5 24 53 pm

screen shot 2016-12-07 at 5 24 17 pm

screen shot 2016-12-07 at 5 24 01 pm

### CODE REVIEW

**Comment by sg00dwin** [comment]
@spadgett PTAL

**Comment by sg00dwin** [comment]
@spadgett updated pr, I went through each page and zeroed out bottom margin of last ui elements to enable a consistent bottom margin across all pages.

I found a few unused classes and removed those. 
A couple of pages the annotations where not within resource-details so moved those. 

Fixes openshift#994
Fixes openshift#1002
Fixes openshift#993

**Comment by spadgett** [comment]
@rhamilto Do you mind reviewing this one?

@sg00dwin Thanks. This is a nit, but I think a slightly smaller bottom margin might be better?

File: app/styles/_substructure.less [hunk 1]
```diff
@@ -74,12 +79,20 @@ html,body {
...
+          // default 50px space beneath content
+          padding-bottom: @middle-content-bottom-margin;
           position: relative;
           width: 100%;
         }
       }
     }
   }
+  & .overview {
+    .middle .middle-section .middle-container .middle-content {
```
Comment (rhamilto) [inline app/styles/_substructure.less]: Nit:  no need to nest this rule.

File: app/styles/_substructure.less [hunk 2]
```diff
@@ -265,3 +278,34 @@ html,body {
     }
   }
 }
+
+
+// Removal of varying bottom-margin from ui elements that are positioned last within the
+// middle-content area in order to achieve consistent bottom spacing across all pages
+.middle-content {
+  .table:last-child {
```
Comment (rhamilto) [inline app/styles/_substructure.less]: Nit:  alphabetize rules in this section by selector?

File: app/styles/_tables.less [hunk 3]
```diff
@@ -210,3 +211,7 @@
     overflow-wrap: normal;
   }
 }
+
+.table-top-margin {
```
Comment (rhamilto) [inline app/styles/_tables.less]: Is there an advantage to adding this rule over using the existing spacer classes?

File: app/views/browse/build-config.html [hunk 4]
```diff
@@ -432,11 +432,11 @@ <h3 class="hidden visible-lg visible-xl">Details</h3>
                   <uib-tab heading="Environment" active="selectedTab.environment" ng-if="buildConfig && !(buildConfig | isJenkinsPipelineStrategy)">
                     <uib-tab-heading>Environment</uib-tab-heading>
                     <h3>Environment Variables</h3>
-                    <div style="margin-top: -5px;" ng-if="BCEnvVarsFromImage.length">
+                    <p ng-if="BCEnvVarsFromImage.length">
```
Comment (rhamilto) [inline app/views/browse/build-config.html]: Hooray for more semantic markup!

**Comment by sg00dwin** [comment]
@rhamilto @spadgett  updated pr changes per comments

**Comment by spadgett** [comment]
[merge]

**Pull request merged**